functions/add_recommendations/main.py

```diff
@@ -0,0 +1,46 @@
+"""Cloud Function -- Adding Song Recommendations from ListenBrainz (BigQuery) + Genius API"""
+import pandas as pd
+import pandas_gbq as pd_gbq
+from utilities import bigquery, common, genius
+
+def get_song_metadata(info):
+    """Get (combined) top-hit song info via Genius API using ListenBrainz info
+    """
+    top_hit = genius.get_top_hit(info[0], info[1])
+    if not top_hit:
+        return {}
+    hit_artist, hit_title = common.d_apostrophe(
+        [top_hit['primary_artist']['name'], top_hit['title']]
+    )
+    if (info[0].lower() != hit_artist.lower() or info[1].lower() != hit_title.lower()):
+        return {}
+    hit_song_id = top_hit['id']
+    song = genius.get_song(hit_song_id)
+    return {
+        'album': song['album']['name'] if song['album'] else song['album'],
+        'apple_music_player_url': song['apple_music_player_url'],
+        'artist': song['primary_artist']['name'],
+        'embed_content': song['embed_content'],
+        'id': song['id'],
+        'song_art_image_url': song['song_art_image_url'],
+        'time_of_day': info[3],
+        'title': song['title'],
+        'url': song['url'],
+    }
+
+def entry_point(request): # pylint: disable=unused-argument
+    """Cloud Function Entry Point
+    """
+    df = pd.DataFrame( # pylint: disable=invalid-name
```

Comment:
avoid abbreviation
https://google.github.io/styleguide/pyguide.html#3161-names-to-avoid